qiskit/transpiler/runningpassmanager.py

```diff
@@ -118,6 +119,22 @@ def run(self, circuit, output_name=None, callback=None):
         else:
             circuit.name = name
         circuit._layout = self.property_set['layout']
+        for gate_name, cals in calibration.items():
+            if isinstance(cals, dict):
+                for qubit_params, sched in cals.items():
+                    circuit.add_calibration(
+                        gate=gate_name,
+                        qubits=qubit_params[0],
+                        schedule=sched,
+                        params=qubit_params[1],
+                    )
+            else:
+                circuit.add_calibration(
+                    gate=gate_name,
+                    qubits=list(cals.keys())[0][0],
```

Comment:
If Cals is not a five, why can you call keys?

Reply:
This is simply copying the calibrations from a circuit to its transpiled circuit. It would have have been more clear if `circuit.calibrations` had a setter method instead but I feel like that's exactly what `circuit.add_calibration` do but with params sent separately.

Reply:
Right, my question is why we can call "keys" if Cals is not a dictionary (as the else of the logic would imply)